patches/tModLoader/Terraria/Program.TML.cs

```diff
@@ -120,6 +120,44 @@ private static void SetSavePath()
 			Logging.tML.Info($"Controlled Folder Access feature detected. If game fails to launch make sure to add \"{Environment.ProcessPath}\" to the \"Allow an app through Controlled folder access\" menu found in the \"Ransomware protection\" menu."); // Before language is loaded, no need to localize
 	}
 
+	private static void StartupSequenceTml(bool isServer)
+	{
+		try {
+			ControlledFolderAccessSupport.CheckFileSystemAccess();
+			Logging.Init(isServer ? Logging.LogFile.Server : Logging.LogFile.Client);
+
+			if (Platform.Current.Type == PlatformType.Windows && System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture != System.Runtime.InteropServices.Architecture.X64)
+				throw new Exception("The current Windows Architecture of your System is CURRENTLY unsupported. Aborting...");
```

Comment:
Can just be a `FatalExit`, rather than exception that's immediately caught. We could probably split `LogStartup` to at least add env info to the logs, but probably not necessary, this is fine.